src/test/java/com/fasterxml/jackson/core/testutil/failure/JacksonTestFailureExpectedInterceptor.java

```diff
@@ -29,6 +29,39 @@ public void interceptTestMethod(Invocation<Void> invocation,
         handleUnexpectePassingTest(invocationContext);
     }
 
+    /**
+     * Interceptor API method that is called to intercept test template
+     * method invocation, like {@link org.junit.jupiter.params.ParameterizedTest}.
+     *<p>
+     * Before failing passing test case, it checks if the test should pass according to the
+     * {@link ExpectedPassingTestCasePredicate} provided in the
+     * {@link JacksonTestFailureExpected} annotation.
+     */
+    @Override
+    public void interceptTestTemplateMethod(Invocation<Void> invocation,
+            ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext)
+        throws Throwable
+    {
+
+        try {
+            invocation.proceed();
+        } catch (Throwable t) {
+            // do-nothing, we do expect an exception
+            return;
+        }
+
+        // Before we fail the test for passing,
+        // check if the test should pass according to the predicate given
+        ExpectedPassingTestCasePredicate predicate = findzExpectedPassingTestCasePredicate(invocationContext);
+        if (predicate != null) {
+            if (predicate.shouldPass(invocationContext.getArguments())) {
+                // do-nothing, we do not expect an exception
+                return;
+            }
+        }
```

Comment:
What we are trying to do here is use `ExpectedPassingTestCasePredicate` to allow passing test being part of test cases of provided by `@ParameterizedTest` this way we can keep test cases in one place (like `@MethodSource`)